src/OpenTelemetry/Resources/OtelServiceNameEnvVarDetector.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="OtelServiceNameEnvVarDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Resources
+{
+    internal class OtelServiceNameEnvVarDetector : IResourceDetector
+    {
+        private const string EnvVarKey = "OTEL_SERVICE_NAME";
+
+        public Resource Detect()
+        {
+            var resource = Resource.Empty;
+
+            string envResourceAttributeValue = Environment.GetEnvironmentVariable(EnvVarKey);
```

Comment:
Might want to mirror this `SecurityException` logic here:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/54d46d80a03acfa0813021e5875a00a7ef423274/src/OpenTelemetry.Exporter.Jaeger/JaegerExporterOptions.cs#L54

Reply:
Will do

Reply:
Addressed e0a9aa40